modules/solid_mechanics/src/materials/ADComputeIncrementalShellStrain.C

```diff
@@ -410,17 +461,65 @@ ADComputeIncrementalShellStrain::computeGMatrix()
       //  the strain tensor (with g_1, g2_, g_3 base vectors) to
       //  the stress tensor (with base vectors e1, e2, e3)
 
-      (*_ge[j])[i](0, 0) = gi0 * e1;
-      (*_ge[j])[i](0, 1) = gi0 * e2;
-      (*_ge[j])[i](0, 2) = gi0 * e3;
-      (*_ge[j])[i](1, 0) = gi1 * e1;
-      (*_ge[j])[i](1, 1) = gi1 * e2;
-      (*_ge[j])[i](1, 2) = gi1 * e3;
-      (*_ge[j])[i](2, 0) = gi2 * e1;
-      (*_ge[j])[i](2, 1) = gi2 * e2;
-      (*_ge[j])[i](2, 2) = gi2 * e3;
+      (*_ge[j])[i](0, 0) = gi0 * _e1;
+      (*_ge[j])[i](0, 1) = gi0 * _e2;
+      (*_ge[j])[i](0, 2) = gi0 * _e3;
+      (*_ge[j])[i](1, 0) = gi1 * _e1;
+      (*_ge[j])[i](1, 1) = gi1 * _e2;
+      (*_ge[j])[i](1, 2) = gi1 * _e3;
+      (*_ge[j])[i](2, 0) = gi2 * _e1;
+      (*_ge[j])[i](2, 1) = gi2 * _e2;
+      (*_ge[j])[i](2, 2) = gi2 * _e3;
+    }
+  }
+}
+
+void
+ADComputeIncrementalShellStrain::computeLocalCoordinate()
+{
+  // default option:the convention used to calculate the local nodal coordinates is also adopted for
+  // calculating the element's local coordinates
+  // the first local vector can be also defined by the user
+
+  // All nodes have the same normal vector. Therefore, here, we use only the normal vecor of the
+  // first element as "the element's normal vector"
+  _e3 = _node_normal[0];
+
+  if (_is_v1_defined)
+  {
+    _e1 = getParam<RealVectorValue>("first_local_vector_ref");
+
+    if (MooseUtils::absoluteFuzzyEqual(_e1.norm(), 0.0, 1e-3))
```

Comment:
This check can be moved to the class's constructor so you just need to do it once up front.